### PR TITLE
Add release workflow and Sonar in `main` branch

### LINKED ISSUE #44: Add release workflow and Sonar in `main` branch
### Description

Add a release workflow triggering from manual dispatch. This workflow'll responsible for:

* Publish a new release with the distributables.
* Generate a tag

Besides that, we need to run Sonar analysis when a pull request is merged to `main` branch.

### PR DESCRIPTION
**Pull request by jorgegarciarey**
Closes #44

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  GO_VERSION: 1.21.5
+
+jobs:
+  verify:
+    if: github.event.pull_request.merged == true
+    name: Verify
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
```
Comment (m4rii0) [inline .github/workflows/main.yml:19]: We can switch to the v4 version now :)

```suggestion
        uses: actions/checkout@v4
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        with:
+          fetch-depth: 0
+
+      - name: Setup Go version
+        uses: actions/setup-go@v4
```
Comment (m4rii0) [inline .github/workflows/main.yml:24]:
```suggestion
        uses: actions/setup-go@v5
```

Ditto here

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  GO_VERSION: 1.21.5
+  DIST_DIR: dist
+
+jobs:
+  release:
+    name: Release
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
```
Comment (m4rii0) [inline .github/workflows/release.yml:22]:
```suggestion
        uses: actions/checkout@v4
```

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        with:
+          fetch-depth: 0
+
+      - name: Setup Go tools
+        uses: actions/setup-go@v4
```
Comment (m4rii0) [inline .github/workflows/release.yml:27]:
```suggestion
        uses: actions/setup-go@v5
```

**Review by m4rii0 — CHANGES_REQUESTED** [review]

File: version
Comment (hielfx) [inline version:1]: We need this file for now.

**Review by m4rii0 — DISMISSED** [review]

**Pull request merged**